src/lib/y2network/type_detector.rb

```diff
@@ -0,0 +1,141 @@
+# Copyright (c) [2019] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "y2network/interface_type"
+
+module Y2Network
+  # Detects type of given interface. New implementation of what was in
+  # @see Yast::NetworkInterfaces.GetType
+  class TypeDetector
+    class << self
+      # Finds type of given interface
+      #
+      # @param iface [String] interface name
+      #
+      # @return [Y2Network::InterfaceType, nil] type of given interface or nil if cannot be recognized
+      def type_of(iface)
+        type_by_sys(iface) || type_by_config(iface) || nil
```

Comment:
NP: would still try to infer the type from the ifcfg-name (ifcfg-dummy0 or ifcfg-ib0 are good examples of interface where could be useful)

Reply:
@mchf it is something for you. Please consider when you get back. I will merge it for now to prevent more collisions

Reply:
there is a hook for infering type from name here: https://github.com/yast/yast-network/pull/927/files#diff-0901740064f05799d30009695e74334dR449
 It currently considers only loopback's ifcfg file but can be extended easily when needed.